statsmodels/tsa/vector_ar/tests/test_var_jmulti.py

```diff
@@ -0,0 +1,577 @@
+    for ds in datasets:
+        for dt in dt_s_list:
+            if debug_mode:
+                print("\n" + dt_s_tup_to_string(dt) + ": ", end="")
+            lags = results_ref[ds][dt]["whiteness"]["tested order"]
+
+            obtained = results_sm[ds][dt].test_whiteness_new(nlags=lags)
+            # test statistic
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "TEST STATISTIC")
+            desired = results_ref[ds][dt]["whiteness"]["test statistic"]
+            yield assert_allclose, obtained.test_statistic, desired, \
+                rtol, atol, False, err_msg
+            # p-value
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "P-VALUE")
+            desired = results_ref[ds][dt]["whiteness"]["p-value"]
+            yield assert_allclose, obtained.pvalue, desired, \
+                rtol, atol, False, err_msg
+
+            obtained = results_sm[ds][dt].test_whiteness_new(nlags=lags,
+                                                             adjusted=True)
+            # test statistic (adjusted Portmanteau test)
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "TEST STATISTIC (ADJUSTED TEST)")
+            desired = results_ref[ds][dt]["whiteness"]["test statistic adj."]
+            yield assert_allclose, obtained.test_statistic, desired, \
+                rtol, atol, False, err_msg
+            # p-value (adjusted Portmanteau test)
+            err_msg = build_err_msg(ds, dt, "WHITENESS OF RESIDUALS - "
+                                            "P-VALUE (ADJUSTED TEST)")
+            desired = results_ref[ds][dt]["whiteness"]["p-value adjusted"]
+            yield assert_allclose, obtained.pvalue, desired, \
+                rtol, atol, False, err_msg
+
+
+def test_exceptions():
+    if debug_mode:
+        if "exceptions" not in to_test:
+            return
+        else:
+            print("\n\nEXCEPTIONS\n", end="")
+    for ds in datasets:
+        for dt in dt_s_list:
+            if debug_mode:
+                print("\n" + dt_s_tup_to_string(dt) + ": ", end="")
+
+            # instant causality:
+            ### 0<signif<1
+            yield assert_raises, ValueError,\
+                results_sm[ds][dt].test_inst_causality, \
+                0, 0  # this means signif=0
+            ### causing must be int, str or iterable of int or str
+            yield assert_raises, TypeError,\
+                results_sm[ds][dt].test_inst_causality, [0.5]  # 0.5 not an int
+
+
+def setup():
+    datasets.append(macro)  # TODO: append more data sets for more test cases.
+
+    for ds in datasets:
+        load_data(ds, data)
+        results_ref[ds] = load_results_jmulti(ds, dt_s_list)
+        results_sm[ds] = load_results_statsmodels(ds)
+        return results_sm[ds], results_ref[ds]
```

Comment:
Actually the `return` is unnecessary. The important task in the `setup` function is to fill the (global) `result_XXX` dictionaries.
In this case the return did no harm because we fill the dictionaries with the results for only _one_ dataset - so the `return` is not stopping the loop prematurely.
I will delete the `return` statement with the next commit. Thanks for making me think about it!